### PR TITLE
#36265 store smart documents generated document in temp folder

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/autoconfigure/SmartDocumentsAutoConfiguration.kt [hunk 1]
```diff
@@ -158,4 +160,13 @@ class SmartDocumentsAutoConfiguration {
     ): SmartDocumentsDemoResource {
         return SmartDocumentsDemoResource(pluginService, runtimeService)
     }
+
+    @Bean
+    @ConditionalOnMissingBean(ThreadPoolTaskScheduler::class)
+    fun threadPoolTaskScheduler(): ThreadPoolTaskScheduler {
+        val threadPoolTaskScheduler = ThreadPoolTaskScheduler()
+        threadPoolTaskScheduler.poolSize = 5
+        threadPoolTaskScheduler.threadNamePrefix = "ThreadPoolTaskScheduler"
```
Comment (theo-ritense) [inline document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/autoconfigure/SmartDocumentsAutoConfiguration.kt:169]: Shouldn't we give this a better name?

File: document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/client/SmartDocumentsClient.kt [hunk 2]
```diff
@@ -50,6 +69,38 @@ class SmartDocumentsClient(
...
+    ): FileStreamResponse {
+        try {
+            val bodyFlux = webClient().post()
+                .uri("/wsxmldeposit/deposit/unattended")
+                .contentType(APPLICATION_JSON)
+                .bodyValue(smartDocumentsRequest)
+                .retrieve()
+                .bodyToFlux<DataBuffer>()
+
+            val tempFile = Files.createTempFile("smartDocumentsResponse", ".tmp")
```
Comment (theo-ritense) [inline document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/client/SmartDocumentsClient.kt:84]: I believe the story mentioned a 'Temporary file storage solution needs to be created as a separate "feature" built in core(?)'. I don't think this is what was intended.

File: document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/autoconfigure/SmartDocumentsAutoConfiguration.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/autoconfigure/SmartDocumentsAutoConfiguration.kt]: changed

File: document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/client/SmartDocumentsClient.kt (hunk 2 above)
Comment (Klaas-Ritense) [reply document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/client/SmartDocumentsClient.kt]: done

**Review by marijnritense — APPROVED** [review]

**Pull request merged**